File: CP(Competetive Programming)/0.Basic/Sorting Algorithm/1.Bubble Sort/BubbleSort.java
public class BubbleSort {
    public static void main(String[] args) {
        int arr[] = {5, 2, 6, 1, 3, 4};
        /*BubbleSort obj = new BubbleSort();
        obj.Sort(arr, );*/
        for(int i = arr.length-1; i>=0; i--){
            for(int j = i-1; j>=0; j--){
                if(arr[j]>arr[i]){
                    int temp = arr[j];
                    arr[j] = arr[i];
                    arr[i] = temp;
                }
            }
        }
        for(int i = 0; i<arr.length; i++){
            System.out.println(arr[i]);
        }
    }
}

File: CP(Competetive Programming)/0.Basic/Sorting Algorithm/1.Bubble Sort/Bubble Sort.cpp
#include<iostream>
using namespace std;

void BubbleSort(int A[], int n){
    int i, j, temp;
    //int count = 0;

    for(i=0; i<n; i++){
        for(j=0; j<n-i-1; j++){
            //count++;
            if(A[j] > A[j+1]){
                temp = A[j];
                A[j] = A[j+1];
                A[j+1] = temp;
            }
        }
    }
    //cout<<count<<endl;
}

int main(){
    int A[] = {3, 4, 7, 2, 6, 1};
    BubbleSort(A, 6);
    for(int k = 0; k<6; k++){
        cout<< A[k] << " ";
    }
    return 0;
}
